app/src/test/java/io/hstream/testing/ClusterKillNodeTest.java

```diff
@@ -46,18 +68,40 @@ public void teardown() throws Exception {
   }
 
   @Test
-  void listStreamAfterKillNode() {
+  void listStreamAfterKillNodes() {
+    String stream = randStream(hStreamClient);
     hServers.get(0).close();
-    hStreamClient.listStreams();
+    hServers.get(1).close();
+    Assertions.assertEquals(stream, hStreamClient.listStreams().get(0).getStreamName());
+  }
+
+  @Test
+  void killAllShouldFail() throws Exception {
+    for (int i = 0; i < 3; i++) {
+      terminateHServerWithLogs(0, i);
+    }
+    Assertions.assertThrows(
+        Exception.class,
+        () -> {
+          hStreamClient.listStreams();
+        });
   }
 
   @Test
-  // @RepeatedTest(5)
-  void write() {
+  void listSubscriptionAfterKillNodes() {
+    String stream = randStream(hStreamClient);
+    String subscription = randSubscription(hStreamClient, stream);
+    hServers.get(0).close();
+    hServers.get(1).close();
+    Assertions.assertEquals(
+        subscription, hStreamClient.listSubscriptions().get(0).getSubscriptionId());
+  }
+
+  @RepeatedTest(5)
+  void write() throws Exception {
     var streamName = TestUtils.randText();
-    System.out.println("hserver cluster size is " + hServers.size());
+    System.out.println("[DEBUG]: HServer cluster size is " + hServers.size());
```

Comment:
I strongly, maybe not now, but still, encourage you to use a logging framework to do such jobs.

Reply:
056f050